### ISSUE TITLE
Bevy 0.10 dependency causes error?

### ISSUE THREAD

**Issue by pberck** [jakerr/bevy_mod_kira#1]
Hi,

I think bevy_mod_kira is not compatible with bevy 0.11. Pulling it in gives me an error similar to [https://github.com/bevyengine/bevy/issues/9274](https://github.com/bevyengine/bevy/issues/9274). It appears to be because bevy_mod_kira is depending on bevy 0.10.

**Comment by jakerr** [comment]
Hi, thanks for the report!

I should have some time to look at this tomorrow.

**Comment by jakerr** [comment]
I've mostly finished the upgrade in the branch: [next](https://github.com/jakerr/bevy_mod_kira/compare/next)

There's one outstanding issue which is when trying to use multiple file format features the asset loader crashes.
```
cargo run --release --features=ogg,mp3,wav,flac --example file_formats
```

The crash:

```
thread 'IO Task Pool (0)' panicked at /Users/jakerr/.cargo/registry/src/index.crates.io-6f17d22bba15001f/bevy_asset-0.11.3/src/asset_server.rs:265:52:
index out of bounds: the len is 7 but the index is 7
   0: rust_begin_unwind
             at /rustc/187b8131d4f760f856b214fce34534903276f2ef/library/std/src/panicking.rs:597:5
   1: core::panicking::panic_fmt
             at /rustc/187b8131d4f760f856b214fce34534903276f2ef/library/core/src/panicking.rs:72:14
   2: core::panicking::panic_bounds_check
             at /rustc/187b8131d4f760f856b214fce34534903276f2ef/library/core/src/panicking.rs:190:5
   3: bevy_asset::asset_server::AssetServer::get_asset_loader
   4: bevy_asset::asset_server::AssetServer::get_path_asset_loader
   5: async_task::raw::RawTask<F,T,S,M>::run
   6: <futures_lite::future::Or<F1,F2> as core::future::future::Future>::poll
   7: async_executor::Executor::run::{{closure}}
   8: futures_lite::future::block_on
note: Some details are omitted, run with `RUST_BACKTRACE=full` for a verbose backtrace.
stack backtrace:
   0: rust_begin_unwind
             at /rustc/187b8131d4f760f856b214fce34534903276f2ef/library/std/src/panicking.rs:597:5
   1: core::panicking::panic_fmt
             at /rustc/187b8131d4f760f856b214fce34534903276f2ef/library/core/src/panicking.rs:72:14
   2: core::panicking::panic_bounds_check
             at /rustc/187b8131d4f760f856b214fce34534903276f2ef/library/core/src/panicking.rs:190:5
   3: bevy_asset::thread 'IO Task Pool (1)' panicked at /Users/jakerr/.cargo/registry/src/index.crates.io-6f17d22bba15001f/bevy_asset-0.11.3/src/asset_server.rs:asset_server265::AssetServer::get_asset_loader:52:

index out of bounds: the len is 7 but the index is 7
   4: bevy_asset::asset_server::AssetServer::get_path_asset_loader
   5: async_task::raw::RawTask<F,T,S,M>::run
   6: <futures_lite::future::Or<F1,F2> as core::future::future::Future>::poll
   7: async_executor::Executor::run::{{closure}}
   8: futures_lite::future::block_on
note: Some details are omitted, run with `RUST_BACKTRACE=full` for a verbose backtrace.
stack backtrace:
   0: rust_begin_unwind
             at /rustc/187b8131d4f760f856b214fce34534903276f2ef/library/std/src/panicking.rs:597:5
   1: core::panicking::panic_fmt
             at /rustc/187b8131d4f760f856b214fce34534903276f2ef/library/core/src/panicking.rs:72:14
   2: core::panicking::panic_bounds_check
             at /rustc/187b8131d4f760f856b214fce34534903276f2ef/library/core/src/panicking.rs:190:5
   3: bevy_asset::asset_server::AssetServer::get_asset_loader
   4: bevy_asset::asset_server::AssetServer::get_path_asset_loader
   5: async_task::raw::RawTask<F,T,S,M>::run
   6: <futures_lite::future::Or<F1,F2> as core::future::future::Future>::poll
   7: async_executor::Executor::run::{{closure}}
   8: futures_lite::future::block_on
note: Some details are omitted, run with `RUST_BACKTRACE=full` for a verbose backtrace.
```
I won't have time dig further until Monday or so.

**Comment by pberck** [comment]
Thanks!
I'll only need .wav files at the moment so maybe the error above doesn't affect me.
I'll give it a try in the weekend!

EDIT: compiles now, need to experiment a bit.

**Comment by jakerr** [comment]
The crash experienced with bevy 0.11.0 seems to be fixed upstream by upgrading to 0.12.0.
I've released a new `0.2.0` of this crate to crates.io with that updated dependency.
